### ISSUE TITLE
[Feature]: `kubectl cnpg logs pretty`

### ISSUE THREAD

**Issue by leonardoce** [cloudnative-pg/cloudnative-pg#5770]
### Is there an existing issue already for this feature request/idea?

- [X] I have searched for an existing issue, and could not find anything. I believe this is a new feature request to be evaluated.

### What problem is this feature going to solve? Why should it be added?

The output of the CNPG Pods is a machine-readable JSON stream, but unfortunately is hard on the human eyes.
As an example:

```
$ kubectl cnpg logs cluster cluster-example | head
{"level":"info","ts":"2024-10-09T07:05:41Z","logger":"setup","msg":"Starting CloudNativePG Instance Manager","logging_pod":"cluster-example-3","version":"1.24.0","build":{"Version":"1.24.0-dev122","Commit":"f18665bac","Date":"2024-10-08"}}
{"level":"info","ts":"2024-10-09T07:05:41Z","logger":"setup","msg":"Checking for free disk space for WALs before starting PostgreSQL","logging_pod":"cluster-example-3"}
{"level":"info","ts":"2024-10-09T07:05:41Z","logger":"setup","msg":"starting tablespace manager","logging_pod":"cluster-example-3"}
{"level":"info","ts":"2024-10-09T07:05:41Z","logger":"setup","msg":"starting external server manager","logging_pod":"cluster-example-3"}
{"level":"info","ts":"2024-10-09T07:05:41Z","logger":"setup","msg":"starting controller-runtime manager","logging_pod":"cluster-example-3"}
{"level":"info","ts":"2024-10-09T07:05:41Z","msg":"Starting EventSource","controller":"instance-cluster","controllerGroup":"postgresql.cnpg.io","controllerKind":"Cluster","source":"kind source: *v1.Cluster"}
{"level":"info","ts":"2024-10-09T07:05:41Z","msg":"Starting Controller","controller":"instance-cluster","controllerGroup":"postgresql.cnpg.io","controllerKind":"Cluster"}
{"level":"info","ts":"2024-10-09T07:05:41Z","msg":"Starting EventSource","controller":"instance-database","controllerGroup":"postgresql.cnpg.io","controllerKind":"Database","source":"kind source: *v1.Database"}
{"level":"info","ts":"2024-10-09T07:05:41Z","msg":"Starting Controller","controller":"instance-database","controllerGroup":"postgresql.cnpg.io","controllerKind":"Database"}
{"level":"info","ts":"2024-10-09T07:05:41Z","msg":"Starting webserver","logging_pod":"cluster-example-3","address":"localhost:8010","hasTLS":false}
```

It would be great to have a tool allowing the user to get human-readable logs, possibly familiar to the eyes of a PostgreSQL's DBA.

### Describe the solution you'd like

I'd like to have a `kubectl cnpg logs decode` command allowing me to decode the JSON stream and get something more readable.
That command could also allow me to filter only for the loggers I'm interested in. Like in:

```
$ kubectl logs cluster-example-1 | kubectl cnpg logs decode --filter postgresql --level info
```

Or even:

```
$ kubectl cnpg  logs cluster cluster-example -f | kubectl cnpg logs decode --filter postgresql --level info
```

### Describe alternatives you've considered

Manually decoding the logs with some magic helper tool, like `jq`.
That is indeed possible and would cover this use case, too, but would it be nice to have something ready to use?

### Backport?

No

### Are you willing to actively contribute to this feature?

Yes

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
